### PR TITLE
*(ticdc): support health API

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: ref #4757 

### What is changed and how it works?

Implement `api/v1/health`. Make the API reflects the health status of the TiCDC cluster.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?

No

##### Do you need to update user documentation, design documentation or monitoring documentation?

No

### Release note 

```release-note
Make the API `api/v1/health` reflects the health status of the TiCDC cluster.
```

### CODE REVIEW

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/owner/owner.go [hunk 1]
```diff
@@ -593,6 +598,23 @@ func (o *ownerImpl) handleQueries(query *Query) error {
...
+		for _, cfReactor := range o.changefeeds {
+			provider := cfReactor.GetInfoProvider()
+			if provider == nil || !provider.IsInitialized() {
+				// The scheduler has not been initialized yet, it is considered
+				// unhealthy, because owner can not schedule tables for now.
+				isHealthy = false
+				break
+			}
+		}
+		if !o.changefeedTicked {
```
Comment (sdojjy) [inline cdc/owner/owner.go:612]: check `changefeedTicked` first?

**Review by sdojjy — APPROVED** [review]

**Pull request merged**